src/p2p/p2p_protocol_defs.h

```diff
@@ -440,6 +440,14 @@ namespace nodetool
 #endif
 
 
+  inline crypto::hash get_proof_of_trust_hash(const nodetool::proof_of_trust& pot)
+  {
+    std::string s;
+    s.append(reinterpret_cast<const char*>(&pot.peer_id), sizeof(pot.peer_id));
```

Comment:
Yeah, at a guess this is broken for big-endian boxes. The only thing I know of that's big-endian these days is SPARC, and I no longer have access to any.